### DISCUSSION TITLE
Get the fontend sef link to the item in the backend edit view of that item

### DISCUSSION THREAD

**Discussion by laurentgarin** [joomengine/Joomla-Component-Builder#650] [Q&A] [answered]
Let me explain. I have created a custom component, when editing the element and saving it once, I would like to display next to the alias the sef frontend link to this element. 

Here is a screenshot to explain : https://i.stack.imgur.com/HKyt2.jpg

If someone could just give me some guidance on the best way to do this it would be great :-)

**Comment by Llewellynvdm** [comment, marked as answer]
Please keep in account that the SEF system in Joomla has various layers of outcome effects. Please watch this [tutorial](https://www.youtube.com/watch?v=tgtonbyyjDs&list=PLQRGFI8XZ_wtGvPQZWBfDzzlERLQgpMRE&index=96) to understand what is possible in JCB.

Once you understand how this work, you will see that some values may not yet be know, or be ready at the time the item is made. So you will first need to check those values, and with that be able to give instruction to the user so they can set those values. Then once you have the correct values then you can with the [following option](https://www.youtube.com/watch?v=hh4IIPmYIY8&list=PLQRGFI8XZ_wtGvPQZWBfDzzlERLQgpMRE&index=69) add links to the list view, like we are doing in the list views of some JCB views, like the admin list view.

**Reply by Llewellynvdm** [reply]
So yes I did not explain how the code should look like, just mainly what you should first take into account. But I would just use the same code which I will use on the front to setup the links, I would load the front HelperRoute class and use it in the back. Let me know if you are unable to achieve this.
